examples/tests/python/example_test/tensorflow_custom_op/test_zero_out_op.py

```diff
@@ -12,6 +15,7 @@
 # This code is from the guide in https://www.tensorflow.org/guide/extend/op.
 class ZeroOutTest(tf.test.TestCase):
 
+  @unittest.skipIf(sys.version_info[0:2] == (3, 7), "See https://github.com/pantsbuild/pants/issues/7417.")
```

Comment:
Can this be made into a decorator in `test_base.py`? Not requiring the use of `TestBase` (since we can't here), but perhaps something like:
```python
def skip_for_interpreter_versions(versions, msg):
  return unittest.skipIf(any('.'.join(sys.version_info[0:2]) == version for version in versions), msg)
```

Reply:
Otherwise it's not entirely clear at a glance that what we're doing here is skipping a specific interpreter.

Reply:
Opened https://github.com/pantsbuild/pants/pull/7420 as a precursor to this PR being merged.

Reply:
Hm. Between #7420 and this line, I think I'd just prefer this line, unless we're expecting a lot more usage.

Reply:
I agree with you @stuhood. 

I don't anticipate us in the future ever wanting to intentionally skip a test because it does not work with the specified interpreter—that would mean for new features, we are actively saying "this does not work on every Python version we claim Pants supports, and we're (temporarily) okay with that." We should _not_ accept PRs that do not work fully for 2.7, 3.6, and 3.7.

This evokes John's dislike for the `@flaky_test` decorator. We do not want to make it convenient and easy to introduce bad practices, like a test not working for a certain interpreter.

--

For this particular instance, this line will also go away very soon thanks to Danny's patch https://github.com/pantsbuild/pants/pull/7424.